### PR TITLE
Rename canonical-schema to canonical-data.schema.json

### PR DESCRIPTION
**Pull request by ErikSchierboom**
While going over old issues, I came across https://github.com/exercism/problem-specifications/issues/700

That issue suggest to rename `canonical-schema.json` to `canonical-data.schema.json`, which is a sensible change I feel.
Therefore, this PR updates the filename as suggested.

Closes https://github.com/exercism/problem-specifications/issues/700

### CODE REVIEW

**Review by SaschaMann — APPROVED** [review]

**Review by junedev — APPROVED** [review]

**Review by BethanyG — APPROVED** [review]

**Review by wolf99 — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@iHiD Could you take a look? We need an exercism/maintainers-admin approval.

**Review by iHiD — APPROVED** [review]

**Pull request merged**
